CDCgov/prime-reportstream
# Update project to Java 17

This PR contains changes needed to build the local project in Java 17. It also incorporates changes to the CI/CD pipeline made in #10931 and updates to a newer copy of the SonarCloud scanner action.

Test Steps:
1. Ensure a local build using Java language level 17 is successful and all tests pass.

## Changes
- Updates to `prime-router` to support builds targeting Java 17.
- Updates to CI/CD pipeline to target Java 17.

## Checklist

### Testing

## Linked Issues
- #10719
- #10815
